test/functional/wallet_importdescriptors.py

```diff
@@ -687,11 +689,15 @@ def run_test(self):
         descriptor["timestamp"] = 0
         descriptor["next_index"] = 0
 
-        batch = []
-        batch.append(encrypted_wallet.walletpassphrase.get_request("passphrase", 3))
-        batch.append(encrypted_wallet.importdescriptors.get_request([descriptor]))
+        encrypted_wallet.walletpassphrase("passphrase", 99999)
+        t = threading.Thread(target=encrypted_wallet.importdescriptors, args=([descriptor],))
+
+        with self.nodes[0].assert_debug_log(expected_msgs=[f'Rescan started from block 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206... (slow variant inspecting all blocks)'], timeout=5):
+            t.start()
+
+        self.nodes[0].cli('-rpcwallet="encrypted_wallet" walletlock')
```

Comment:
Oh yes, `walletlock` never gets called here. However, implementing this change results in another `CannotSendRequest`.

Reply:
'assert_raises_rpc_error' could be used here, but wouldn't that introduce another race condition where this assertion will fail if the rescan has already been completed?

Reply:
Yes, I have implemented this both for `walletlock` and `walletpassphrasechange` to ensure that this error is raised in both. However, it is important to note that this test now tests something slightly different than what it did previously. Previously this checked that the wallet was still able to complete a full rescan even if an attempt was made to relock the wallet (because of the timeout in `walletpassphrase`). This new test does not test this because other than using the `walletpassphrase` I can't think of another method of deleting the passphrase from memory without having an error thrown. That being said, this adds coverage for something that was not previously tested (encryption RPCs throwing errors during a rescan, also introduced in #26347 ) and removes the race condition, so I think this is an improvement.

Reply:
I have now added test coverage for the lock as well.